edge/pkg/devicetwin/dtmanager/membership.go

```diff
@@ -295,7 +295,8 @@ func removeDevice(context *dtcontext.DTContext, toRemove []dttype.Device, baseMe
 		deleteResult := dttype.MembershipUpdate{BaseMessage: baseMessage, RemoveDevices: RemoveDevices}
 		result, err := dttype.MarshalMembershipUpdate(deleteResult)
 		if err != nil {
-
+			klog.Errorf("Remove device %s failed, marshal membership err: %s", device.ID, err)
+			continue
 		} else {
```

Comment:
Looks like we don't need `else` to judge

Reply:
yes, just remove else

Reply:
Done